source/derived/mus_auxFieldVar_module.fpp

```diff
@@ -2180,6 +2182,67 @@ contains
+  !! in the auxField for the incompressible fluid model.
+  !! Reference:
+  !! 1) Zhaoli Guo and T. S. Zhao. “Lattice Boltzmann Model for Incompressible
+  !!    Flows through Porous Media”. In: Physical Review E 66.3 (2002), p. 036304.
+  !!    doi: 10.1103/PhysRevE.66.036304.
+  !! 2) Irina Ginzburg. “Consistent lattice Boltzmann schemes for the Brinkman
+  !!    model of porous flow and infinite Chapman-Enskog expansion”. In: Phys.
+  !!    Rev. E 77 (6 June 2008), p. 066704. doi: 10.1103/PhysRevE.77.066704.
+  subroutine mus_addBrinkmanToAuxField_fluidIncomp(fun, auxField, iLevel,     &
+    &                                               time, varSys, phyConvFac, &
+    &                                               derVarPos)
+    ! ------------------------------------------------------------------------ !
+    !> Description of method to update source
+    class(mus_source_op_type), intent(inout) :: fun
+    !> output auxField array
+    real(kind=rk), intent(inout)         :: auxField(:)
+    !> current level
+    integer, intent(in)                :: iLevel
+    !> current timing information
+    type(tem_time_type), intent(in)    :: time
+    !> variable system definition
+    type(tem_varSys_type), intent(in) :: varSys
+    !> Physics conversion factor for current level
+    type(mus_convertFac_type), intent(in) :: phyConvFac
+    !> position of derived quantities in varsys
+    type(mus_derVarPos_type), intent(in) :: derVarPos(:)
+    ! ------------------------------------------------------------------------ !
+    integer :: vel_pos(3), elemOff
+    integer :: iElem, nElems, posInTotal
+    real(kind=rk) :: bCoeffField(fun%elemLvl(iLevel)%nElems), velFac
+    ! ------------------------------------------------------------------------ !
+    ! position of velocity field in auxField
+    vel_pos = varSys%method%val(derVarPos(1)%velocity)%auxField_varPos(1:3)
+    ! Number of elements to apply source terms
+    nElems = fun%elemLvl(iLevel)%nElems
+    ! Get the force variable from the configuration for each element
+    call varSys%method%val(fun%data_varPos)%get_valOfIndex( &
+      & varSys  = varSys,                                   &
+      & time    = time,                                     &
+      & iLevel  = iLevel,                                   &
+      & idx     = fun%elemLvl(iLevel)%idx(1:nElems),        &
+      & nVals   = nElems,                                   &
+      & res     = bCoeffField                               )
+
+!$omp parallel do schedule(static), private( posInTotal, elemOff )
+    !NEC$ ivdep
+    do iElem = 1, nElems
+      posInTotal = fun%elemLvl(iLevel)%posInTotal(iElem)
+      ! element offset
+      elemoff = (posInTotal - 1) * varSys%nAuxScalars
+      velFac = 1.0_rk / ( 1.0_rk + bCoeffField(ielem) / (2 * phyConvFac%sourceCoeff) )
+
+      ! add force to velocity
+      auxfield(elemoff + vel_pos) = auxfield(elemoff + vel_pos) * velFac
```

Comment:
Sorry, I used the short term here merely out of lazyness. Please unroll the vel_pos field here, and denote the three individual scalings accordingly.
Otherwise it may be treated as a loop, and might misplace vectorization. Having the three individual entries may be more beneficial.

Reply:
Yes, exactly. It hopefully allows it to vectorize over that longer loop. Otherwise it may be tempted to only vectorize the short 3 element loop.